pkg/cfn/builder/vpc.go

```diff
@@ -35,10 +41,29 @@ func (c *ClusterResourceSet) addSubnets(refRT *gfn.Value, topology api.SubnetTop
 			SubnetId:     refSubnet,
 			RouteTableId: refRT,
 		})
+
+		if api.IsEnabled(c.spec.VPC.AutoAllocatedIPv6) {
+			c.newResource(alias+"CIDRv6", &gfn.AWSEC2SubnetCidrBlock{
+				SubnetId:      refSubnet,
+				Ipv6CidrBlock: getCIDRv6(counter),
+			})
+		}
+
 		c.subnets[topology] = append(c.subnets[topology], refSubnet)
+		counter++
 	}
 }
 
+func getCIDRv6(n int) *gfn.Value {
```

Comment:
What is `n` here? Can we rename it to something more explicit?

Reply:
This is a bit obscure if you don't know what those `MakeFn...()` do. Can you add a comment of what it returns? :/

Reply:
It slices the AWS-provided IPv6 CIDR and returns Nth element... comment would be good, for sure.

Reply:
How about `selectIPv6Element(index int)`? Or something like that?

Reply:
Let me try a different thing, I'm not sure how useful this function is actually, I might just inline this in the other function.

Reply:
Oh, I just saw it. I think it was better here, with the same comment as the one you added inlined. Mainly, because if it's an implementation detail of CF it's nicer to abstract it a little and give it a name.